### PR TITLE
Add unittesting support to .codeclimate.yml

### LINKED ISSUE #465: Add unittesting support to .codeclimate.yml
#### Issue Summary

The codeclimate.yml file has the ability to tell CodeClimate what we're using for unittests. Please add this to the .yml file.

### PR DESCRIPTION
**Pull request by prashuchaudhary**
closes #465

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=471) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-python?pullRequest=471) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-python?pullRequest=471) it.</sub>

**Comment by mbernier** [comment]
What do we need to do for the token?

**Comment by prashuchaudhary** [comment]
@mbernier We have to generate it

**Comment by prashuchaudhary** [comment]
@mbernier For an organization-owned repository, one must be in an organization's Owners group to generate the token. Alternatively, we can add Open Source Repo but for that one must have GitHub-rights to administer the repository.

**Comment by prashuchaudhary** [comment]
@mbernier Unfortunately, codeclimate.com is under maintenance. So can't do anything.

**Comment by mbernier** [comment]
Ahh, ok - I got confused with the codecov stuff here. I will make sure to get the code and add it to our Travis environment. 

Can you please update the config to pull a CODE_CLIMATE_TOKEN variable from the travis environment?

**Comment by prashuchaudhary** [comment]
@mbernier update the code to pull CODE_CLIMATE_TOKEN variable from the travis env

**Comment by mbernier** [comment]
Perfect! As soon as I can get into CodeClimate I will get this fixed up. Thanks!!

**Comment by mbernier** [comment]
I have added the token to travis! Let's run this thing one more time - merging master to cause that.

**Comment by mbernier** [comment]
One more merge to try to get codeclimate to see our tests!

**Comment by mbernier** [comment]
@prashuchaudhary it looks like there's an issue:
```
$ ./cc-test-reporter after-build --exit-code $TRAVIS_TEST_RESULT
ERRO[0000] failed to read file __init__.py
open __init__.py: no such file or directory 
Error: open __init__.py: no such file or directory
Usage:
  cc-test-reporter after-build [flags]
Flags:
  -s, --batch-size int               batch size for source files (default 500)
  -e, --coverage-endpoint string     endpoint to upload coverage information to (default "https://api.codeclimate.com/v1/test_reports")
  -t, --coverage-input-type string   type of input source to use [simplecov, lcov, coverage.py, clover, gocov, gcov, cobertura, jacoco]
      --exit-code int                exit code of the test run
  -r, --id string                    reporter identifier
  -p, --prefix string                the root directory where the coverage analysis was performed (default "/home/travis/build/sendgrid/sendgrid-python")
Global Flags:
  -d, --debug   run in debug mode
```

File: .travis.yml [hunk 1]
```diff
@@ -23,10 +26,14 @@ addons:
...
 - . ./test/prism.sh
 - prism version
+- curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./cc-test-reporter
+- chmod +x ./cc-test-reporter
+- ./cc-test-reporter before-build
 script:
 - if [[ $TRAVIS_PYTHON_VERSION == '2.6' ]]; then coverage run -m unittest2 discover; else coverage run -m unittest discover; fi
 after_script:
 - codecov
+- ./cc-test-reporter after-build --exit-code $TRAVIS_TEST_RESULT
```
Comment (mbernier) [inline .travis.yml]: @prashuchaudhary it looks like there's an issue:
```
$ ./cc-test-reporter after-build --exit-code $TRAVIS_TEST_RESULT
ERRO[0000] failed to read file __init__.py
open __init__.py: no such file or directory 
Error: open __init__.py: no such file or directory
Usage:
  cc-test-reporter after-build [flags]
Flags:
  -s, --batch-size int               batch size for source files (default 500)
  -e, --coverage-endpoint string     endpoint to upload coverage information to (default "https://api.codeclimate.com/v1/test_reports")
  -t, --coverage-input-type string   type of input source to use [simplecov, lcov, coverage.py, clover, gocov, gcov, cobertura, jacoco]
      --exit-code int                exit code of the test run
  -r, --id string                    reporter identifier
  -p, --prefix string                the root directory where the coverage analysis was performed (default "/home/travis/build/sendgrid/sendgrid-python")
Global Flags:
  -d, --debug   run in debug mode
```

Comment (prashuchaudhary) [reply .travis.yml]: will check and update

Comment (mbernier) [reply .travis.yml]: Thank you!

Comment (mbernier) [reply .travis.yml]: @prashuchaudhary please let me know when you get a chance to look at this and I will loop back around.

**Comment by prashuchaudhary** [comment]
@mbernier creating a new pull request which is updated with remote master and closing this.
https://github.com/sendgrid/sendgrid-python/pull/537

**Pull request closed without merging**